apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClientPartial.cs

```diff
@@ -85,10 +89,93 @@ public partial class BigtableServiceApiSettings
                 retryFilter: IdempotentRetryFilter
             );
 
+        // TODO: Validate range
+        // TODO: Link to snippet for creating GcpCallInvoker manually from here.
+        /// <summary>
+        /// The maximum number of channels than will be open concurrently to the Bigtable endpoint.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Note that channels are managed and shared amongst <see cref="BigtableClient"/> instances using the same
+        /// endpoint and channel options and which use the default credentials. So this property will have no bearing
+        /// across clients using different options. To create a custom grouping of channels to be managed, create a
+        /// <see cref="GcpCallInvoker"/> manually and use the
+        /// <see cref="BigtableClient.Create(CallInvoker, BigtableServiceApiSettings, string)">BigtableClient.Create</see>
+        /// overload taking a <see cref="CallInvoker"/> to create clients from it.
+        /// </para>
+        /// </remarks>
+        public int MaxChannels { get; set; } = 20;
```

Comment:
@jskeet The recommendation from the Bigtable team is to cap this at 4 channels per CPU. It seems a bit odd though to initialize this to something like `Environment.ProcessorCount * 4` though. I would think we'd probably want to pick some constant default but create some docs around this. Thoughts?

Reply:
I *think* I'd be okay using the processor count - the thread pool used to be set up that way by default, for example. It's obviously not as simple as just using a hard-coded value though. (I'd probably default to 16, so assuming 4 processors.)

Reply:
I'm confused. So are you saying you prefer using the processor count * 4 or using hard-coded 16? I'll use 16 for now but let me know if I'm reading this wrong.